### PR TITLE
Ensure SubDownloadLink is present

### PR DESCRIPTION
**Pull request by Zillode**

### CODE REVIEW

File: lib/suby/downloader/opensubtitles.rb [hunk 1]
```diff
@@ -30,7 +30,8 @@ class Downloader::OpenSubtitles < Downloader
 
     def download_url
       SEARCH_QUERIES_ORDER.find(lambda { raise NotFoundError, "no subtitles available" }) { |type|
-        if subs = search_subtitles(search_query(type))['data']
+        subs = search_subtitles(search_query(type))['data']
+        if subs.length != 0 && subs.first['SubDownloadLink']
```
Comment (eregon) [inline lib/suby/downloader/opensubtitles.rb]: `!subs.empty?` would be more idiomatic I think.

**Comment by eregon** [comment]
There is some problem with the Travis build, which seems unrelated, I'll fix it.

File: lib/suby/downloader/opensubtitles.rb [hunk 2]
```diff
@@ -30,7 +30,8 @@ class Downloader::OpenSubtitles < Downloader
 
     def download_url
       SEARCH_QUERIES_ORDER.find(lambda { raise NotFoundError, "no subtitles available" }) { |type|
-        if subs = search_subtitles(search_query(type))['data']
+        subs = search_subtitles(search_query(type))['data']
+        if !subs.empty? && subs.first['SubDownloadLink']
```
Comment (eregon) [inline lib/suby/downloader/opensubtitles.rb]: this removed the `nil` check, so Travis fails partly because of this :wink:

**Comment by Zillode** [comment]
I'm actually using https://github.com/Zillode/suby/commit/962804000f8c61f63d6265d7662bd52cd996450d but that changes the current behavior.

**Comment by eregon** [comment]
I see, I found a a couple more things to fix locally.
Thank you for your patch!

**Comment by eregon** [comment]
Ahah, I like the `sample` idea, with a bit of luck it picks different ones over time and hopefully at some point the right one :)

**Comment by eregon** [comment]
I integrated your idea in 23496ff19707d6239fafb294531374ea319c810f

**Pull request merged**